packages/manager/src/features/Account/utils.ts

```diff
@@ -17,6 +18,26 @@ interface GetRestrictedResourceText {
   resourceType: GrantTypeMap;
 }
 
+interface GrantsProfileSchema {
+  grants: Grants | undefined;
+  profile: Profile | undefined;
+}
+
+interface AccountAccessGrant extends GrantsProfileSchema {
+  globalGrantType: 'account_access';
+  permittedGrantLevel: GrantLevel;
+}
+
+interface NonAccountAccessGrant extends GrantsProfileSchema {
+  globalGrantType: Exclude<GlobalGrantTypes, 'account_access'>;
+  permittedGrantLevel?: GrantLevel;
+}
+
+// Discriminating union to determine the type of global grant
+export type IsRestrictedGlobalGrantType =
```

Comment:
```suggestion
export type RestrictedGlobalGrantType =
```
Do we need the `Is`?